### PR TITLE
Registration confirmation

### PR DESCRIPTION
**Pull request by AndrewTehan**
# Description

[Added registration confirmation via email. Added confirmation endpoint](https://trello.com/c/r7MwgMmk)
## Type of change

- [ ] Bug fix
- [x] New feature 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by nieisnuje — COMMENTED** [review]
update your branch with new code from 'authentication-registration' branch

File: app/views/user_mailer/registration_confirmation.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+  </head>
+
+   <body>
+    <h1>Hi, <%= @user.full_name %>!!!</h1>
+    <p>
+      Thanks for registering! To confirm your registration click the URL below.
+      <%# confirm_email_user_url(@user.confirm_token) %>
+      <%= "http://localhost:3000/api/v1/confirmations?confirm_token=#{@user.confirm_token}&id=#{@user.id}"%>
```
Comment (nieisnuje) [inline app/views/user_mailer/registration_confirmation.html.erb]: It's not ok))

Comment (nieisnuje) [reply app/views/user_mailer/registration_confirmation.html.erb]:
```
<%= "#{@host}/api/v1/confirmations?confirm_token=#{@user.confirm_token}&id=#{@user.id}"%>

def host # you can put it into helper
  Rails.env.production? ? 'heroku link' : 'http://localhost:3000'
end
```

File: app/views/user_mailer/registration_confirmation.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+  </head>
+
+   <body>
+    <h1>Hi, <%= @user.full_name %>!!!</h1>
+    <p>
+      Thanks for registering! To confirm your registration click the URL below.
+      <%# confirm_email_user_url(@user.confirm_token) %>
```
Comment (nieisnuje) [inline app/views/user_mailer/registration_confirmation.html.erb]: remove comments if you don't need them

File: app/views/user_mailer/registration_confirmation.html.erb (hunk 1 above)
Comment (AndrewTehan) [reply app/views/user_mailer/registration_confirmation.html.erb:14]: Ok) should I make a prefix for the path to the endpoint (something like it `new_user_session_path`) or something else?

Comment (nieisnuje) [reply app/views/user_mailer/registration_confirmation.html.erb]: idk
I think this url is ok
just adjust it according to the environment

File: app/api/api.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      end
+    end
+
+    def authenticate!
+      error!('401 Unauthorized', 401) unless current_user
+    end
+
+    def host
+      Rails.env.production? ? 'morning-plains-26206.herokuapp.com' : 'http://localhost:3000'
+    end
```
Comment (nieisnuje) [inline app/api/api.rb]: would be better if you created helpers folder and put this into host_helper.rb or something like it

**Review by alhimick-prog — APPROVED** [review]

**Review by nieisnuje — APPROVED** [review]

**Pull request merged**